staff/lab/last-lab

```diff
@@ -1,19 +1,82 @@
 #!/usr/bin/env python3
 import argparse
 import functools
+import re
 import sys
+import textwrap
+from datetime import date
+from datetime import datetime
+from datetime import time
+from datetime import timedelta
 
 import ocflib.lab.stats
 from ocflib.infra.hosts import hostname_from_domain
 from tabulate import tabulate
 
 
-def make_fqdn(s):
+def make_fqdn(hostname):
     """Argparse custom type that appends ".ocf.berkeley.edu" to short hostnames."""
-    if '.' not in s:
-        return s + '.ocf.berkeley.edu'
+    if '.' not in hostname:
+        return hostname + '.ocf.berkeley.edu'
     else:
-        return s
+        return hostname
+
+
+def parse_datetime(dtstr):
+    """Argparse custom type for parsing time expresssions."""
+    # TODO: consider switching to dateparser once that becomes feasible
```

Comment:
Why is using dateparser not feasible? Do we need all these formats, or can we just have one standard one to make things easier?

Reply:
++ on using some pre-existing library for this. we don't need to handle the exact same date formats that `last` accepts. this is an awful lot of unnecessary date-parsing logic.

Reply:
There's currently no Debian package for dateparser or dateutils, so unless we create a venv for utils or package them ourselves, it's not usable here.

I've tried parsedatetime too, but both that and dateparser give pathological results for some common date formats:

```
>>> import parsedatetime
>>> cal = parsedatetime.Calendar()
>>> cal.parse("20180223101845")  # This is one of `last`'s formats
(time.struct_time(tm_year=2018, tm_mon=2, tm_mday=23, tm_hour=22, tm_min=9, tm_sec=40, tm_wday=4, tm_yday=54, tm_isdst=0), 0)

>>> import dateparser
>>> dateparser.parse("20180223101845")
datetime.datetime(2033, 12, 12, 9, 45, 10)
```

What do you think? Should we limit it to one fixed format (plus "now" and "today")? Should we try to rope in one of the libraries anyway?

Reply:
https://packages.debian.org/stretch/python3-dateutil ?

just require yyyy-mm-dd imo, this should be plenty flexible enough

Reply:
I think you might want to do something like `yyyy-mm-dd hh:mm` so that times can be allowed (we don't have second-based granularity anyway, so I'd just do to the minute), but I agree that just having one format would work fine as long as it's specified in the help output and is made clear when it doesn't work what the format should be. I don't think we really need to support all the possible formats if we don't have to.